### PR TITLE
Add more cauldron behavior

### PR DESCRIPTION
**Pull request by good777LUCKY**
Added Behavior:
  - Leather armors can be undyed with cauldron that filled with water. (Partially add #659)
  - Can remove the top-most pattern layer of a banner with cauldron that filled with water. (Add #660)

New Method:
  - ItemBanner.hasPattern()

### CODE REVIEW

File: src/main/java/cn/nukkit/item/ItemBanner.java [hunk 1]
```diff
@@ -76,6 +76,10 @@ public int getPatternsSize() {
         return (this.hasCompoundTag() ? this.getNamedTag() : new CompoundTag()).getList("Patterns").size();
     }
 
+    public boolean hasPattern() {
```
Comment (RORIdev) [inline src/main/java/cn/nukkit/item/ItemBanner.java:79]: Please add `PowerNukkitOnly` and `Since` attributes

**Review by RORIdev — CHANGES_REQUESTED** [review]
Thanks for contributing! Please address the following requests.

File: src/main/java/cn/nukkit/item/ItemBanner.java (hunk 1 above)
Comment (good777LUCKY) [reply src/main/java/cn/nukkit/item/ItemBanner.java:83]: Okay

**Review by RORIdev — APPROVED** [review]
Thanks!

**Review by RORIdev — APPROVED** [review]

**Pull request merged**